sdk/servicebus/service-bus/src/core/messageReceiver.ts

```diff
@@ -711,6 +713,34 @@ export class MessageReceiver extends LinkEntity {
     };
   }
 
+  /**
+   * Prevents us from receiving any further messages.
+   */
+  public stopReceivingMessages(): Promise<void> {
+    log.error(`[${this._receiver?.name}]: User has requested we stop receiving new messages`);
+    this._stopReceivingMessages = true;
+
+    return this.drainReceiver();
+  }
+
+  private drainReceiver() {
+    const drainPromise = new Promise<void>((resolve) => {
+      if (this._receiver == null) {
+        resolve();
+        return;
+      }
+
+      this._receiver.once(ReceiverEvents.receiverDrained, () => {
+        resolve();
+      });
+
+      this._receiver.drain = true;
+      this._receiver.addCredit(1);
```

Comment:
I would recommend adding a few log statements here as well. One for when we enable draining and another for when we receive the drain event

Reply:
Yeah, that's a great point, I'll add that in (especially since we suspect this can occasionally be a hang).

Reply:
Added some logging. Also swapped over to log.receiver to match the other logging in the file.